pkgs/os-specific/linux/intel-sgx/default.nix

```diff
@@ -0,0 +1,296 @@
+      cp ${build}/sgx_sign $out/bin/
+      cp ${build}/sgx_encrypt $out/bin/
+      cp ${build}/gdb-sgx-plugin/sgx-gdb $out/bin/
+
+      # link sdk folder (several tools expect this exact folder structure)
+      SDK_DIR=$out/usr/share/sgxsdk
+      mkdir -p $SDK_DIR
+      cp common/buildenv.mk $SDK_DIR/
+
+      ln -s $out/include $SDK_DIR/include
+      ln -s $out/lib64 $SDK_DIR/lib64
+
+      mkdir -p $SDK_DIR/bin
+      ln -s $out/bin $SDK_DIR/bin/x86
+      ln -s $out/bin $SDK_DIR/bin/x64
+    '' else ''
+      # copy service binaries
+      mkdir -p $out/aesm
+      cp ${build}/libipc.so $out/aesm/
+      cp ${build}/liboal.so $out/aesm/
+      cp ${build}/libutils.so $out/aesm/
+      cp ${build}/libdcap_quoteprov.so $out/aesm/
+      cp ${build}/libsgx_default_qcnl_wrapper.so $out/aesm/
+      cp ${build}/liburts_internal.so $out/aesm/
+      cp ${build}/libCppMicroServices.so.4.0.0 $out/aesm/
+      cp ${build}/aesm_service $out/aesm/
+
+      # signed binaries
+      cp ${build}/le_prod_css.bin $out/aesm/
+      cp ${build}/libsgx_le.signed.so $out/aesm/
+      cp ${build}/libsgx_pve.signed.so $out/aesm/
+      cp ${build}/libsgx_qe.signed.so $out/aesm/
+      cp ${build}/libsgx_pce.signed.so $out/aesm/
+
+      cp external/dcap_source/QuoteGeneration/psw/ae/data/prebuilt/libsgx_qe3.signed.so $out/aesm/
+      cp external/dcap_source/QuoteGeneration/build/linux/libsgx_pce_logic.so $out/aesm/
+      cp external/dcap_source/QuoteGeneration/build/linux/libsgx_qe3_logic.so $out/aesm/
+
+      # copy bundle binaries
+      mkdir -p $out/aesm/bundles
+      cp ${build}/bundles/libepid_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/lible_launch_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/liblinux_network_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libpce_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libepid_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libecdsa_quote_service_bundle.so $out/aesm/bundles/
+      cp ${build}/bundles/libquote_ex_service_bundle.so $out/aesm/bundles/
+
+      mkdir -p $out/aesm/data
+      cp psw/ae/aesm_service/data/white_list_cert_to_be_verify.bin $out/aesm/data/
+
+      # render: copy shared libraries
+      mkdir -p $out/lib64
+      cp ${build}/libsgx_uae_service.so $out/lib64/
+      cp ${build}/libsgx_epid.so $out/lib64/libsgx_epid.so.1
+      cp ${build}/libsgx_launch.so $out/lib64/libsgx_launch.so.1
+      cp ${build}/libsgx_quote_ex.so $out/lib64/libsgx_quote_ex.so.1
+      cp ${build}/libsgx_urts.so $out/lib64/
+      cp ${build}/libsgx_enclave_common.so $out/lib64/libsgx_enclave_common.so.1
+
+      # install provided udev rules
+      mkdir -p $out/etc/udev/rules.d
+      cp linux/installer/common/libsgx-enclave-common/91-sgx-enclave.rules $out/etc/udev/rules.d/
+      cp linux/installer/common/sgx-aesm-service/92-sgx-provision.rules $out/etc/udev/rules.d/
+    '';
```

Comment:
I like the idea to compile and run some of the SDK samples in simulation mode in the `installCheckPhase`. That way we can have some confidence the SDK is working.

https://github.com/yaxitech/nix-linux-sgx/blob/7aea4d11910d011b37a3f5bc1daa9307a86b15c2/pkgs/intel-sgx/sdk.nix#L205-L241